cmd/commands/cmd_payments.go

```diff
@@ -473,6 +483,26 @@ func SendPayment(ctx *cli.Context) error {
 
 	req.PaymentAddr = payAddr
 
+	if ctx.IsSet("route_hints") {
```

Comment:
We should allow route_hints always if NO payment_request was provided, because we also wanna use it via AMP Sends not only keysends.

Reply:
there's an exit early upstream if pay_req is set, I left the logic to error if neither are set (a more explicit code pathway). Hope that's okay.

Reply:
Route hinting did not appear to work with standard payments in my testing, hence why I restricted it to AMP/keysend. @MPins felt that it should not be added to AMP to avoid conflicts for when they are embedded in the AMP request (similar to pay_req). I'm new to lightning dev so I'm not familiar with all the nuances of the protocol.

Reply:
So in general we should only prevent it if there is a reason for it, right now how you implemented it, it would accept AMP spontaneous payments, normal AMP payments (without a payment request) and keysend non-AMP payments. So the only thing which would be missing are legacy payments or MPP payments which specify a payment hash and destination manually (tho they would likely fail on the receiver side if a corresponding invoice for this hash) can be found. So in any way I think we can just add the route_hints here because the later cash is very unlikely to happen and should not be a problem for LND to handle it properly.

Would be interesting which problems you had enabling it for everything.

Reply:
I used the itest harness to recreate the various payment scenarios, its been a while now I can't remember all the various things I tried but it would just hang indefinitely and eventually fail with no route found for standard payments (normal and legacy). It passed for both keysend and amp which is why I had enabled it for those cases.

My guess is it's a rather safe thing to have enabled across all payment types and if there does happen to be a scenario where it's not working, someone can file an issue. I will just remove any restriction.